timesketch/api/v1/resources/searchtemplate.py

```diff
@@ -51,4 +85,100 @@ def get(self):
         Returns:
             View in JSON (instance of flask.wrappers.Response)
         """
-        return self.to_json(SearchTemplate.query.all())
+        templates = SearchTemplate.query.all()
+        meta = {
+            'collection': []
+        }
+        for template in templates:
+            saved_searches = View.query.filter_by(
+                searchtemplate=template)
+            for saved_search in saved_searches:
+                data = {
+                    'search_id': saved_search.id, 'template_id': template.id}
+                meta['collection'].append(data)
+
+        return self.to_json(templates, meta=meta)
+
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+
+        Returns:
+            View in JSON (instance of flask.wrappers.Response)
+        """
+        form = request.json
+        if not form:
+            form = request.data
+
+        search_id = form.get('search_id')
+        if not search_id:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Unable to save the searchtemplate, the search has not '
+                'been saved first')
+
+        sketch_id = form.get('sketch_id')
+        if not sketch_id:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'We need to have a sketch associated with the saved search '
```

Comment:
You should get the sketch_id from the saved search object, right? A saved search isn't possible to save without a sketch attached.

Reply:
yes, that's true... I should just pull it from there instead of requesting it again...